python/cudf/cudf/dataframe/series.py

```diff
@@ -1069,6 +1072,32 @@ def masked_assign(self, value, mask):
         data = self._column.masked_assign(value, mask)
         return self._copy_construct(data=data)
 
+    def drop_duplicates(self, keep="first", inplace=False):
+        """
+        Return Series with duplicate values removed
+        """
+        in_cols = [self._column]
+        in_index = self.index
+        from cudf.dataframe.multiindex import MultiIndex
+        if isinstance(in_index, MultiIndex):
+            in_index = RangeIndex(len(in_index))
+        out_cols, new_index = cpp_drop_duplicates(
+            [in_index.as_column()], in_cols, None, keep
+        )
+        new_index = as_index(new_index)
+        if len(self.index) == len(new_index) and self.index.equals(new_index):
```

Comment:
I believe this length check is already done in the `self.index.equals` call so there's no need to do it here for optimization.

Reply:
```suggestion
        if self.index.equals(new_index):
```

Reply:
@kkraus14, done. Removed the length check.